frontend/public/components/modals/toleration-update-modal.tsx

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import * as React from 'react';
+import { PromiseComponent, Dropdown } from '../utils';
+import {K8sKind, k8sPatch} from '../../module/k8s';
+
+import {createModalLauncher, ModalBody, ModalComponentProps, ModalSubmitFooter, ModalTitle} from '../factory';
+import * as _ from 'lodash-es';
+
+class TolerationUpdateModal extends PromiseComponent {
+  readonly state: TolerationUpdateModalState;
+
+  constructor(public props: TolerationUpdateModalProps) {
+    super(props);
+    this.state.tolerations = this.props.resource.spec.tolerations;
+  }
+
+  _submit = (e) => {
+    e.preventDefault();
+
+    // We just throw away any rows where the key is blank
+    const tolerations = _.reject(this.state.tolerations, t => _.isEmpty(t.key));
+
+    const keys = tolerations.map(t => t.key);
+    if (_.uniq(keys).length !== keys.length) {
+      this.setState({errorMessage: 'Duplicate keys found.'});
+      return;
+    }
+
+    // Convert any blank values to null
+    _.each(tolerations, t => t.value = _.isEmpty(t.value) ? null : t.value);
```

Comment:
Tolerations that use the `Exists` operator will have an empty value. We can't just throw those away.